scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/GlobalFileValidator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.job.factories.spi.model.validator;
+
+import java.net.URI;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.common.exception.PermissionException;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.ModelValidatorContext;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ValidationException;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 27/08/19
+ */
+public class GlobalFileValidator implements Validator<URI> {
+    private static final Logger logger = Logger.getLogger(GlobalFileValidator.class);
+
+    @Override
+    public URI validate(URI parameterValue, ModelValidatorContext context) throws ValidationException {
+        if (context == null || context.getScheduler() == null) {
+            // Sometimes the workflow is parsed and checked without scheduler instance (e.g., submitted from catalog).
+            // In this case, we don't have the access of the scheduler global dataspace, so the validity check is passed.
+            logger.debug(String.format("Can't check the validity of the variable value [%s], because missing the access to the scheduler global data space",
+                                       parameterValue));
+            return parameterValue;
+        }
+
+        // if parameterValue is not a file existing in data space, we throw ValidationException
+        if (StringUtils.isBlank(parameterValue.toString())) {
```

Comment:
Can we use the simple String.isBlack() method instead of involving extra-deps ?

Reply:
I don't find isBlank func in String. (if you mean String.isEmpty(), it doesn't refuse input with only spaces)

Reply:
My bad, isBlank() comes w/ Java 11 !